.github/workflows/create-release-pr.yml

```diff
@@ -0,0 +1,208 @@
+name: Create Release PR
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - release/*
+
+permissions:
+  pull-requests: write
+
+concurrency: "${{ github.workflow }}-${{ github.ref }}"
+
+env:
+  NODE_VERSION: "18.19.1"
+
+jobs:
+  detect-new-version:
+    name: Detect New Version
+    runs-on: ubuntu-latest
+    outputs:
+      NEW_VERSION_TAG: ${{ steps.detect-version.outputs.NEW_VERSION_TAG }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # fetch all history for all tags and branches
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install Yarn
+        run: npm install -g yarn
+
+      - name: Detect New Version
+        id: detect-version
+        run: |
+          NEW_VERSION=$(yarn --silent version:current)
+          git diff origin/main.. -- package.json | grep "\"version\": \"${NEW_VERSION}\"" > /dev/null && HAS_VERSION_CHANGE=0 || HAS_VERSION_CHANGE=1
+          echo "HAS_VERSION_CHANGE=${HAS_VERSION_CHANGE}"
+
+          if [ $HAS_VERSION_CHANGE -eq 0 ]; then
+            echo "New version detected ${NEW_VERSION}"
+
+            echo "IS_NEW_VERSION=true" >> $GITHUB_OUTPUT
+            echo "NEW_VERSION=${NEW_VERSION}" >> $GITHUB_OUTPUT
+            echo "NEW_VERSION_TAG=v${NEW_VERSION}" >> $GITHUB_OUTPUT
+          else
+            echo "No version change detected, exiting"
+            exit 1
+          fi
+
+      - name: Verify Release Changes
+        if: ${{ steps.detect-version.outputs.IS_NEW_VERSION == 'true' }}
+        run: |
+          echo "IS_NEW_VERSION=${{ steps.detect-version.outputs.IS_NEW_VERSION }}"
+          echo "NEW_VERSION=${{ steps.detect-version.outputs.NEW_VERSION }}"
+          echo "NEW_VERSION_TAG=${{ steps.detect-version.outputs.NEW_VERSION_TAG }}"
+
+          function verify_file_has_diff_with_version() {
+            FILE_NAME=$1
+            FILE_DIFF="$(git diff origin/main.. -- ${FILE_NAME})"
+
+            echo "Diff in $FILE_NAME:\n$FILE_DIFF"
+
+            if [ "$(echo "${FILE_DIFF}" | wc -l)" -eq 0 ]; then
+              echo "No diff detected in $FILE_NAME, exiting"
+              exit 1
```

Comment:
if you intend of failing the step as soon as we exit 1, i suggest you add
`set -e` at the start of the script (line 57)

Reply:
in general set -e is a good practice in all bash scripts (unless special circumstances ) 
it will exit as soon as exit code != 0 i received and wont continue

Reply:
added `set -e` in all bash steps, and I'm using `|| true` in specific places.
adjusted flow control accordingly.